src/mono/mono/mini/intrinsics.c

```diff
@@ -83,11 +83,15 @@ mini_emit_inst_for_ctor (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignat
 		return ins;
 
 #ifdef MONO_ARCH_SIMD_INTRINSICS
+#ifdef TARGET_WASM
 	if (cfg->opt & MONO_OPT_SIMD) {
```

Comment:
Why is this needed ?

Reply:
Because we want `WasmBase` intrinsics even when SIMD is not enabled. So I let is pass here and skip the SIMD later in `arch_emit_simd_intrinsics`. Arm has similar intrinsics in ArmBase class, so I took similar approach.

Alternatively I could refactor the code to split SIMD and non-SIMD intrinsics if needed.

Reply:
Refactoring would be better.